app/Main.hs

```diff
@@ -1,13 +1,25 @@
 {-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
 module Main where
 
 import Web.Spock
 import Web.Spock.Config
+import Control.Monad.IO.Class (liftIO)
+import Data.Yaml.Config (loadYamlSettings, useEnv)
+import Data.Aeson (parseJSON, FromJSON, withObject, (.:))
+
+data AppSettings = AppSettings { appPort :: Int }
+
+instance FromJSON AppSettings where
+  parseJSON = withObject "AppSettings" $ \o -> do
```

Comment:
The language extension is `InstanceSigs` if you're interested, but I don't see the point. This isn't a function I've created.